### ISSUE TITLE
Exception caught: ord() expected a character

### ISSUE THREAD

**Issue by pgdr** [FriskByBergen/RPiParticle#83]
The following exception occurred on `sampler.collect()`, but I don't know if it was during testing or actual running:

```python
if ord(s) == SDS011.msg_start:
```

Full log:

Nov 07 21:14:01 raspberrypi fby_client[20889]:

```python
DeviceConfig.logMessage:
{
    "device_id": "FriskPI10",
    "key": "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX",
    "long_msg": "Traceback (most recent call last):
     
    File "/usr/local/friskby/bin/fby_client", line 106, in run
    data = self.collect(sampler)
     
    File "/usr/local/friskby/bin/fby_client", line 77, in collect
    data = sampler.collect( )
     
    File "/usr/local/friskby/lib/sampler.py", line 18, in collect
    values = self.reader.read( )
     
    File "/usr/local/friskby/lib/sds011.py", line 25, in read
    if ord(s) == SDS011.msg_start:
     
    TypeError: ord() expected a character, but string of length 0 found ", "msg":
    "Exception caught: "ord() expected a character, but string of length 0 found"."
}
```

**Comment by joakim-hove** [comment]
I _think_ this is because you are testing while the service is running, then two instances are fighting for the sensor. You need to stop the service before interactive testing.

**Comment by pgdr** [comment]
Are you sure it couldn't also happen if it fails to read a byte?  That is, if `serial.read()` returns `""`?  E.g. if RPi cannot connect to the serial device?

**Comment by joakim-hove** [comment]
No - I am not sure, but I am sure that if you test and run the service concurrently you will get a error like this.

**Comment by pgdr** [comment]
Well — it also happens when I hard-unwire the device (not unplug the USB, but actually disconnect a wire).

When I disconnect the USB, I get the "correct" error:
`Exception in collect device reports readiness to read but returned no data (device disconnected?)`

So there is room for improvements in the `serial.read()` call.  I'll check out what the documentation says.

**Comment by pgdr** [comment]
And also, if I connect it to a different USB port, I get
`serial.serialutil.SerialException: could not open port /dev/ttyUSB0: [Errno 2] No such file or directory: '/dev/ttyUSB0'`
Is it really that sensitive?

**Comment by pgdr** [comment]
Fix proposed #84.

**Comment by pgdr** [comment]
Fixed by #84
